tests/system_tests/test_artifacts/test_wandb_artifacts_api.py

```diff
@@ -241,6 +241,52 @@ def test_log_artifact_with_invalid_tags(tmp_path, user, wandb_init, api, invalid
             run.log_artifact(artifact, tags=invalid_tags)
 
 
+def test_retrieve_artifacts_by_tags(user, wandb_init, server_supports_artifact_tags):
+    project = "test"
+    artifact_name = "test-artifact"
+    artifact_type = "test-type"
+
+    with wandb_init(entity=user, project=project) as run:
+        for i in range(10):
+            artifact = wandb.Artifact(name=artifact_name, type=artifact_type)
+            with artifact.new_file(f"{i}.txt", "w") as f:
+                f.write("testing")
+            run.log_artifact(artifact)
+
+    artifact_name = f"{user}/{project}/{artifact_name}"
+
+    for logged_artifact in Api().artifacts(type_name=artifact_type, name=artifact_name):
+        version = int(logged_artifact.version.strip("v"))
+        if version in [2, 3, 5, 7]:
+            logged_artifact.tags.append("prime")
+        if version % 3 == 0:
+            logged_artifact.tags.append("fizz")
+        if version % 5 == 0:
+            logged_artifact.tags.append("buzz")
```

Comment:
this is amazing and the literal first time I think I've ever seen fizzbuzz in the wild 🔥